luigi/contrib/ecs.py

```diff
@@ -79,19 +79,57 @@ def _get_task_statuses(cluster, task_ids):
     if response['failures'] != []:
         raise Exception('There were some failures:\n{0}'.format(
             response['failures']))
+
     status_code = response['ResponseMetadata']['HTTPStatusCode']
     if status_code != 200:
         msg = 'Task status request received status code {0}:\n{1}'
         raise Exception(msg.format(status_code, response))
 
-    return [t['lastStatus'] for t in response['tasks']]
+    return response['tasks']
+
+def _get_task_statuses(tasks):
+    """
+    Returns list of {RUNNING|PENDING|STOPPED} for each task in tasks
+
+    Args:
+        - tasks (list): list of ECS taskDescriptions
+    """
+    return [t['lastStatus'] for t in tasks]
+
+def _check_exit_codes(tasks, essential_containers):
+    """
+    Checks each essential task in tasks for a failure reason or a
+    non-zero exitCode
 
+    Args:
+        - tasks (list): list of ECS taskDescriptions
+        - essential_containers (list): list of essential container names
 
-def _track_tasks(cluster, task_ids):
+    Raises:
+        - Exception: A failing essential task was found
+    """
+    dirty_exits = []
+    for t in tasks:
+        for container in t['containers']:
+            if container['name'] in essential_containers:
+                if container['lastStatus'] == 'STOPPED':
+                    # Check if container's command returned error
+                    # or if ECS had an error running the command
+                    if 'exitCode' in container and container['exitCode'] != 0 \
+                            or 'reason' in container:
```

Comment:
i imagine that `reasons` are not extraordinarily informative.  but something's better than nothing - maybe we should let them bubble up?